test/boot_arch_test.rb

```diff
@@ -0,0 +1,170 @@
+require_relative "test_helper"
+
+Yast.import "BootArch"
+
+describe Yast::BootArch do
+  subject { Yast::BootArch }
+
+  def stub_arch(arch)
+    Yast.import "Arch"
+
+    allow(Yast::Arch).to receive(:architecture).and_return(arch)
+  end
```

Comment:
for me it is a bit overkill as many methods are just plain if depending on architecture.